### ISSUE TITLE
[BUG] Rancher does not delete psp config upon disabling psp on rancher provisioned RKE1 clusters

### ISSUE THREAD

**Issue by anupama2501** [rancher/rancher#40032]
Information about the Cluster
**Rancher Server Setup**
- Rancher version: `v2.7-head 28a84c7` and `v2.7.0`
- Installation option (Docker install/Helm Chart): `Helm Chart`
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): `RKE1`, `v1.24.9 `, `1.4.3-rc1`
- Proxy/Cert Details: `Self-signed`

**Information about the Cluster**
- Kubernetes version: `v1.24.8`
- Cluster Type (Local/Downstream): `Custom  1 worker, 1 etcd, 1 cp RKE1`
- User: admin/standard user - cluster Owner 

**Describe the bug**
While creating a cluster if we enable PSP and then later edit the config on the cluster and disable PSP, the psp config created by rancher on the RKE1 cluster is not deleted.

**To Reproduce**
1. Create a custom hardened RKE1 cluster by enabling PSP from cluster management page >> Advanced options
2. Once the cluster gets active run `k get psp` the following configs are created:
```
NAME                    PRIV    CAPS   SELINUX    RUNASUSER          FSGROUP     SUPGROUP    READONLYROOTFS   VOLUMES
cis-psp                 true    *      RunAsAny   RunAsAny           RunAsAny    RunAsAny    false            *
default-psp             true    *      RunAsAny   RunAsAny           RunAsAny    RunAsAny    false            *
restricted-noroot-psp   false          RunAsAny   MustRunAsNonRoot   MustRunAs   MustRunAs   false            configMap,emptyDir,projected,secret,downwardAPI,csi,persistentVolumeClaim
```
3. Edit the config on the cluster and disable PSP
4. Verify by running `k get psp` 

**Result**
The PSP configs still exists and are not deleted

**Expected Result**
Expected the psp config to be deleted.

**Additional context**
Happens on 2.7.0 as well.

**Comment by jiaqiluo** [comment]
This is the expected behavior of the current implementation: 
Rancher does not delete the default PSPs when the PSP feature is disabled on the downstream cluster.  They can be deleted by the user if they are no longer needed. 

As the label in the issue indicates, we will add this to the release note. 

Meanwhile, I raise a PR in rancher/docs to reflect the limitation.

We are considering an improvement to address this issue in future releases.

**Comment by snasovich** [comment]
@anupama2501 @markusewalker , are we good to close this one as "functions as designed" (and clear milestone as there is no actual fix in 2.7.2)?

@jiaqiluo , is there docs issue/PR to link here?

**Comment by markusewalker** [comment]
@snasovich that is fine with me. @jiaqiluo linked a doc PR earlier in this issue that makes this note. As such, I'll close this issue as this is the expected behavior and the doc PR has already been merged.
